front/src/modules/pathfinding/helpers/getStepLocation.ts

```diff
@@ -1,19 +1,35 @@
-import type { PathItemLocation } from 'common/api/osrdEditoastApi';
+import type { OperationalPointReference, PathItemLocation } from 'common/api/osrdEditoastApi';
 import { mToMm } from 'utils/physics';
 
 const getStepLocation = (step: PathItemLocation): PathItemLocation => {
+  const trackReference: OperationalPointReference['track_reference'] =
+    !('track' in step) && step.track_reference && 'track_name' in step.track_reference
+      ? { track_name: step.track_reference.track_name }
+      : null;
+
   if ('track' in step) {
     // TODO: step offset should be in mm in the store /!\
     // pathfinding blocks endpoint requires offsets in mm
     return { track: step.track, offset: mToMm(step.offset) };
   }
   if ('operational_point' in step) {
-    return { operational_point: step.operational_point };
+    return { operational_point: step.operational_point, track_reference: trackReference };
```

Comment:
Once you are here, you have already handle the case `'track' in step`, so you can directly use `step.track_reference`
```suggestion
    return { operational_point: step.operational_point, track_reference: step.track_reference };
```

You can do the same below and remove the const `trackReference`

Reply:
Done